extension/src/standalone_pages/InstructorDashboardRenderVideoInfo.js

```diff
@@ -39,4 +75,146 @@ function renderVideoAccordian(document, videoObjects) {
 }
 
-export { renderVideoAccordian };
+function renderActiveFeedback(video, response) {
+    let color= "#5959e6";
+    let activeChart = document.getElementById(`activeFeedback_${video.videoID}`).getContext("2d");
+    response.active_questions.sort((a, b) => {
+        return a.timestamp - b.timestamp;
+    });
+    let labels = response.active_questions.flatMap((question) => {
+        if (question.timestamp !== null && question.timestamp <= video.video_duration) {
+            return [question.timestamp];
+        } else {
+            return [];
+        }
+    });
+    let amounts = response.active_questions.flatMap(question => {
+        if (question.count != null && question.timestamp <= video.video_duration) {
+            return [question.count];
+        } else {
+            return [];
+        }
+    });
+    let questionsMap = new Map();
+    labels.forEach((timestamp) => {
+        questionsMap[timestamp] = response.active_questions_text.filter(question => question.timestamp === timestamp);
+    });
+    let chart = new Chart(activeChart, {
+        type: "bar",
+        data: {
+            labels: labels,
+            datasets: [{
+                label: "Active Feedback",
+                borderColor: color,
+                pointBackgroundColor: color,
+                pointBorderColor: color,
+                pointHoverBackgroundColor:color,
+                pointHoverBorderColor: color,
+                data: amounts,
+            }],
+        },
+        options: {
+            scales: {
+                xAxes: [{
+                    scaleLabel: {
+                        display: true,
+                        labelString: "Timestamp (s)"
+                    },
+                    ticks: {
+                        beginAtZero: true
+                    }
+                }], 
+                yAxes: [{
+                    scaleLabel: {
+                        display: true,
+                        labelString: "# of Questions"
+                    },
+                    ticks: {
+                        beginAtZero: true
+                    }
+                }]
+            }
+        }
+    });
+    let canvas = document.getElementById(`activeFeedback_${video.videoID}`);
+    canvas.onclick = function(evt) {
+        let firstPoint = chart.getElementAtEvent(evt)[0];
+
+        if (firstPoint) {
+            let label = chart.data.labels[firstPoint._index];
+            console.debug(label, questionsMap[label]);
+            displayStudentQuestions(label, questionsMap);
+        }
+    };
+}
+
+function renderPassiveFeedback(video, response) {
+    let color= "#5959e6";
+    console.debug(video);
+    let passiveChart = document.getElementById(`passiveFeedback_${video.videoID}`).getContext("2d");
+    response.passive_question.sort((a, b) => {
+        return a.timestamp - b.timestamp;
+    });
+    let transformedData = response.passive_question.flatMap((question) => { 
```

Comment:
same as above, use `filter` instead of double `flatMap`

Reply:
We need flatmap to reformat the data.
Chart.js requires the data to be in an array [1,2,3,4].
Filter can only eliminate the data and cannot reformat the data from [{timestamp: 1, count: 1}] -> [1]

Also, I'm not sure what you're referring to about "test google classroom"

Reply:
So we actually need to split the data as well.
{timestamp: 1, count: 1}.
I don't believe we can do this with filter and map.
I can try something with lodash.partition but at this point I don't think it's worth it.

I would prefer to re-visit this after the semester is over when I have more time.